torch/csrc/jit/runtime/register_prim_ops_fulljit.cpp

```diff
@@ -1120,45 +1121,103 @@ bool isSortableTupleType(
   return true;
 }
 
-bool isSortableListOfObjectsOrTuples(
+bool isSameTupleOptionalType(const TupleTypePtr& t1, const TupleTypePtr& t2) {
+  auto types1 = t1->elements();
+  auto types2 = t2->elements();
+  size_t n1 = types1.size();
+  size_t n2 = types2.size();
+
+  if (n1 != n2)
+    return false;
+  for (size_t i = 0; i < n1; i++) {
+    auto cur_t1 = types1[i];
+    auto cur_t2 = types2[i];
+
+    if (cur_t1->kind() == TypeKind::NoneType ||
+        cur_t2->kind() == TypeKind::NoneType) {
+      continue;
+    } else if (*cur_t1 != *cur_t2) {
+      return false;
+    }
+  }
+  return true;
+}
+
+// Check whether ivalues are sortable. If so,
+// update comparator if we can get one.
+bool checkAndGetComparator(
```

Comment:
Can you give this a more descriptive name? I think that returning `c10::optional<c10::IValueComparator>` would be clearer too. That would prevent the optimisation of not comparing anything if the list is empty or full of `None`, but I think it's worth it.

Reply:
You could also keep the two steps we have now:
#1 - Call `isSortableListOfObjectsOrTuples`
#2 - If the list is sortable, then get the comparator and sort.